docs/wallets.md

```diff
@@ -0,0 +1,57 @@
+# For Wallet Developers
+
+codex32 is a new format for BIP32 master seeds. It is essentially a replacement for
+BIP39 or SLIP39 seed words, and the user workflow should be much the same, except
+that:
+
+* The character set is different (bech32 characters rather than a wordlist).
+* Seeds may be split across multiple shares, rather than encoded as a single string.
+* It is possible to do specific error detection/correction during entry.
+
+There are two levels of wallet support:
+
+* The ability to import seeds/shares; here we essentially just have recommendations about dealing with errors.
+* The ability to generate seeds/shares on the device; here our guidance is more involved.
+
+We encourage every wallet to support importing seeds and shares, since
+* the technical difficulty is low (roughly on par with that of supporting Segwit addresses, plus optional error-correction support);
+* the added functionality is isolated from the rest of the wallet (once the seed is imported you don't care where it came from); and
+* beyond correctness of the code, there is little risk (no need to source randomness or execute potentially variable-time algorithms).
+
+Supporting seed generation is a little more involved so the tradeoff between
+implementation complexity and user value is less clear, especially since the
+Codex provides users instructions on doing generation themselves.
+
+## Import Support
+
+codex32 shares may be any length between 128 and 512 bits.
+Wallets should support import of 128- or 256-bit seeds; other lengths are optional.
+
+128-bit seeds are 48 characters in length, including the `MS1` prefix.
+256-bit seeds are 74. For other bit-lengths, see the BIP.
+
+The process for entering shares is:
+
+1. The user should enter the first share. To the extent possible given screen limitations, when entering share data, data should be displayed in uppercase, with visually distinct four-character windows. The first four-character window should include the `MS1` prefix, which should be pre-filled.
+1. Once the first share is fully entered, the wallet should validate the checksum and header before accepting it.
+   * The user should not be able to enter mixed-case characters. The user must be able to enter all bech32 characters as well as `?` indicating an erasure. Wallets may allow users to enter non-bech32 characters, at their discretion. (This may be useful to guide error correction, e.g. by attempting to replace all `B`s with `8`s.)
+   * If the header is invalid, the wallet should highlight this and prevent the user from entering additional data until it is fixed. An invalid header is one that starts with a character other than `0` or `2` through `9`, or one which starts with `0` but whose share index is not `S`. For shares after the first, a header is also invalid if its threshold and identifier do not match those of the first share.
```

Comment:
types of errors users make depends on input device.

For example 'B' could be a typo of 'V'. It might make sense for implementations to create a list for erasure correction of invalid characters as some are more likely than others. Similar to how when correcting an invalid share_index, there is no need to check already entered share indices.

Always substituting 'B' with '8' or 'I' with 'L' actually reduces the correction ability since it turns an erasure location into an error location. OTOH when the suggestion is correct, it's more powerful than marking it as an erasure.

Reply:
Besides visual similarity, keyboard proximity, there is also auditory similarity:

When spoken over the phone (and foolishly not using NATO phonetic) which characters are most likely to be mistaken for one another in popular languages.

Having a set of common mistakes (from the above 3 risks) for each bech32 character helps the correction power when an error location is known due to being non-bech32 or some of the other rules. Although easiest implementation may try to correct it as an erasure first, THEN substitute the ? for the top picks based on the character they actually typed to see if they help make it correctable.

Reply:
All good points. I think my current text, which just gives 8 and B as an example, is okay. I can remove the example if you think it's too leading. But I don't want to try to fit a nuanced paragraph about mistake patterns into the spec, since this isn't the right place for it.